src/components/ADempiere/Form/paymentAllocation/tableFrom.vue

```diff
@@ -0,0 +1,161 @@
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <https:www.gnu.org/licenses/>.
+-->
+<template>
+  <div
+    class="card-form"
+    style="height: 100%;overflow: auto;"
+  >
+    <el-table
+      ref="allocationPayments"
+      :data="recordsData"
+      style="width: 100%;height: 100%;"
+      border
+      highlight-current-row
+      show-summary
+      :summary-method="getSummaries"
+      :row-class-name="styleCell ? tableRowClassName : ''"
+      @select="handleSelectionChange"
+    >
+      <el-table-column
+        :type="isSelection ? 'selection' : ''"
+        width="50"
+      />
+      <el-table-column
+        v-for="(valueOrder) in label"
+        :key="valueOrder.columnName"
+        :column-key="valueOrder.columnName"
+        :label="valueOrder.label"
+        width="auto"
+        :min-width="!valueOrder.isNumeric ? valueOrder.size : valueOrder.size"
+        :align="valueOrder.isNumeric ? 'right' : 'left'"
+        :prop="valueOrder.columnName"
+        :sortable="valueOrder.isNumeric"
+      />
+    </el-table>
+  </div>
+</template>
+
+<script>
+import { formatPrice } from '@/utils/ADempiere/valueFormat.js'
+
+export default {
+  name: 'TableFrom',
+  props: {
+    label: {
+      type: Array,
+      default: () => []
+    },
+    recordsData: {
+      type: Array,
+      default: () => []
+    },
+    height: {
+      type: String,
+      default: ''
+    },
+    styleCell: {
+      type: Boolean,
+      default: false
+    },
+    isSelection: {
+      type: Boolean,
+      default: true
+    },
+    selection: {
+      type: Array,
+      default: () => []
+    },
+    addSelection: {
+      type: String,
+      default: ''
+    }
+  },
+  mounted() {
+    this.toggleSelection(this.selection)
+  },
+  methods: {
+    formatPrice,
+    handleSelectionChange(selection, row) {
+      this.$store.dispatch(this.addSelection, selection)
+    },
+    toggleSelection(rows) {
+      if (rows) {
+        rows.forEach(row => {
+          const selectionRow = this.recordsData.find(record => record.NrDocument === row.NrDocument)
+          this.$refs.allocationPayments.toggleRowSelection(selectionRow)
+        })
+      } else {
+        this.$refs.allocationPayments.clearSelection()
+      }
+    },
+    tableRowClassName({ row, rowIndex }) {
+      if (row.Difference > 0) {
+        'success-row'
+      }
+      return 'warning-row'
+    },
+    getSummaries(param) {
```

Comment:
Duplicate method, please add support in another utility file, should be called `numberFormat`, to centralize and import it in your implementations.